src/Test.FunctionalTests.Sarif/TestData/Multitool/ValidateCommand/Inputs/SARIF2004.OptimizeFileSize_Invalid.sarif

```diff
@@ -43,6 +43,36 @@
               }
             }
           ]
+        },
+        {
+          "ruleId": "TEST1001",
+          "ruleIndex": 0,
+          "message": { "text": "Bad result: artifacts array does not provide any additional useful information." },
+          "analysisTarget": { "uri": "local/test/code/src/file1.c" },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "local/test/code/src/file1.c"
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleIndex": -1,
+          "rule": {
+            "id": "TEST1001"
+          },
+          "message": { "text": "Bad result: artifacts array does not provide any additional useful information." }
```

Comment:
>artifacts array does not provide any additional useful information [](start = 44, length = 66)

Comment is wrong: This is invalid because `ruleIndex` and `rule` are both present.

Reply:
1. It is legal to "not have enough information to locate the rule metadata". Rule metadata is entirely optional in SARIF.
2. But in any case, "not enough information to locate the rule metadata" is not one of the cases we are testing for with this new rule. So I don't understand why this test case is here.

---
In reply to: [455257253](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455257253) [](ancestors = 455257253)

Reply:
for this case, we are analyzing the rule where we don't have the ruleIndex AND we don't have the toolComponent. So thats a bad case. With that, i changed to not enough information. Isn't that right?

---
In reply to: [455342702](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455342702) [](ancestors = 455342702,455257253)

Reply:
It is legal not to have rule index. I see now that this test case is "You should use `result.ruleId` instead of `result.rule.id`"

---
In reply to: [455382915](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455382915) [](ancestors = 455382915,455342702,455257253)

Reply:
This comment still doesn't describe this test case. In this test case, `toolComponent` does _not_ refer to the driver, so `result.rule` is necessary. But since `result.rule` is present, `result.ruleId` should _not_ be present. So the message might be "Bad result: ruleId duplicates information in rule."

---
In reply to: [455389470](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455389470) [](ancestors = 455389470,455382915,455342702,455257253)